### ISSUE TITLE
Disable -Wimplicit-fallthrough warnings in FParser code

### ISSUE THREAD

**Issue by jwpeterson** [libMesh/libmesh#2613]
If I recall correctly, this is a GCC-only warning, but when compiling FParser code I see hundreds of lines of warnings:
```
./extrasrc/fp_opcode_add.inc:10662:2: warning: this statement may fall through [-Wimplicit-fallthrough=]
  if(!ByteCodePtr)
  ^~
```
I don't propose trying to actually fix these, but maybe they could be disabled with compiler pragmas.

**Comment by roystgnr** [comment]
This is one of the ones our ignore_warnings.h disables, and the placement of the pragma there says it works for GCC 6 or later.

Of course that doesn't help fparser, which doesn't include any libMesh headers, but we could reasonably make a copy (ideally a stripped down copy, since ignore_warnings.h disables *any* warning that any third-party headers have ever given us) to use there.

**Comment by jwpeterson** [comment]
I'll try just copying the pragma and compiler version checks from ignore_warnings.h

**Comment by jwpeterson** [comment]
Fixed in #2617
